src/chainstate/burn/operations/peg_out_request.rs

```diff
@@ -0,0 +1,256 @@
+        assert_eq!(op.signature.as_bytes(), &signature);
+        assert_eq!(op.amount, amount);
+    }
+
+    #[test]
+    fn test_parse_peg_out_request_should_return_error_given_wrong_opcode() {
+        let mut rng = test_helpers::seeded_rng();
+        let opcode = Opcodes::LeaderKeyRegister;
+
+        let dust_amount = 1;
+        let recipient_address_bytes = test_helpers::random_bytes(&mut rng);
+        let output2 = test_helpers::Output::new_as_option(dust_amount, recipient_address_bytes);
+
+        let mut data = vec![];
+        let amount: u64 = 10;
+        let signature: [u8; 65] = test_helpers::random_bytes(&mut rng);
+        data.extend_from_slice(&amount.to_be_bytes());
+        data.extend_from_slice(&signature);
+
+        let tx = test_helpers::burnchain_transaction(data, output2, opcode);
+        let header = test_helpers::burnchain_block_header();
+
+        let op = PegOutRequestOp::from_tx(&header, &tx);
+
+        match op {
+            Err(OpError::InvalidInput) => (),
+            result => panic!("Expected OpError::InvalidInput, got {:?}", result),
+        }
+    }
+
+    #[test]
+    fn test_parse_peg_out_request_should_return_error_given_no_second_output() {
+        let mut rng = test_helpers::seeded_rng();
+        let opcode = Opcodes::PegOutRequest;
+
+        let output2 = None;
+
+        let mut data = vec![];
+        let amount: u64 = 10;
+        let signature: [u8; 65] = test_helpers::random_bytes(&mut rng);
+        data.extend_from_slice(&amount.to_be_bytes());
+        data.extend_from_slice(&signature);
+
+        let tx = test_helpers::burnchain_transaction(data, output2, opcode);
+        let header = test_helpers::burnchain_block_header();
+
+        let op = PegOutRequestOp::from_tx(&header, &tx);
+
+        match op {
+            Err(OpError::InvalidInput) => (),
+            result => panic!("Expected OpError::InvalidInput, got {:?}", result),
+        }
+    }
+
+    #[test]
+    fn test_parse_peg_out_request_should_return_error_given_no_signature() {
+        let mut rng = test_helpers::seeded_rng();
+        let opcode = Opcodes::PegOutRequest;
+
+        let dust_amount = 1;
+        let recipient_address_bytes = test_helpers::random_bytes(&mut rng);
+        let output2 = test_helpers::Output::new_as_option(dust_amount, recipient_address_bytes);
+
+        let mut data = vec![];
+        let amount: u64 = 10;
+        let signature: [u8; 0] = test_helpers::random_bytes(&mut rng);
+        data.extend_from_slice(&amount.to_be_bytes());
+        data.extend_from_slice(&signature);
+
+        let tx = test_helpers::burnchain_transaction(data, output2, opcode);
+        let header = test_helpers::burnchain_block_header();
+
+        let op = PegOutRequestOp::from_tx(&header, &tx);
+
+        match op {
+            Err(OpError::ParseError) => (),
+            result => panic!("Expected OpError::ParseError, got {:?}", result),
+        }
+    }
```

Comment:
Please also add test coverage for when the OP_RETURN contains bytes beyond the signature.  This should still successfully decode; the extra bytes would be ignored.

Reply:
Done :heavy_check_mark: